### DISCUSSION TITLE
custom resulotion, lora and taesd support

### DISCUSSION THREAD

**Discussion by Amin456789** [LostRuins/koboldcpp#835] [Ideas]
Hi,
thank u for impelenting sd cpp to koboldcpp!

could u please add support for selecting taesd for sd cpp too? taesd makes ram usage and decoding image much much better for cpu users
also i would like to use resulotion such as 384x384 if possible, as it will be much faster and ram friendly, please consider adding custom resulotion
lora support will be great too

kind regards

**Comment by Amin456789** [comment]
sorry for double post,  but i just had an idea and i wanted to share maybe it will be helpful:

for standalone sdui;
1. lora can be loaded via ui in browser instead gui or cmd, for example we can say:
--lora ./lora folder so it can detect all the loras and we can use them in browser instead reloading the gui or cmd all the time by using sdcpp command line which is like <lora:pixelart-lora:1> 

2. and we can run the app without needing the loading of text generation model so all the power can goes to sd cpp image

honestly this way we won't be needing any other gui than koboldcpp for sd cpp too

ive been stalking on updates and i saw u r adding taesd too!  please add this for both chat and standalone sdui, thank u so much for ur hard work, can't wait custom resoultions be added to text generation image chat too

kind regards

**Reply by LostRuins** [reply]
I want to avoid runtime file loading, so most likely I will try for a pre-selected lora option instead.

**Reply by Amin456789** [reply]
thanks for the answer! as a matter fact thats a great idea, i had a second thought andi forgot that we need to reload the model sometimes to clear the lora and use another one so it is better for a whole new new reload again
i  make my own bats so it all takes a double click again
thank u for ur hard work, can't wait to start playing lots of dungeon rpg text games with image generation

**Comment by LostRuins** [comment]
Hi, Custom LORA, TAESD and custom VAE are now supported in the latest version.

**Comment by Amin456789** [comment]
thank u so much for ur hard work, i just deleted the gui i was using for sd cpp too, u rock dude

one problem remains i think = i think after i enabled sdvaeauto every image i make are black with koboldcpp chat [sd ui works great but chat or others dont] almost all images are black specaliy if it is automaticly, maybe it is censorship, but when i type a landscape or a cake the images were being generated with no problem when i typed in custom but lots of other stuff ddint work even in automatic or custom
maybe it is not related to taesd or maybe it is,
im using a lcm model for 1.5

cant wait for custom resolotion too, tnx dude, im ur biggest fan

**Reply by LostRuins** [reply]
could you copy the "generate payload" that appears in the terminal console when the image fails to generate?

**Reply by Amin456789** [reply]
appologize but i dont have such log after generation, it just sais generating image and give the steps and the steps are done [4steps] it shows a black image in the chat. i tried everything even changing almost every settings
btw, it is not the --sdvvaeauto. i tried to run the app without it and still black image is being generated. i tested with quant etc... nothing worked, it was working good before this updtate

**Reply by LostRuins** [reply]
try a different model? I didn't change anything for the image generation itself, so anything that worked before should still work now. also 4 steps may be too little.

**Reply by Amin456789** [reply]
thnxfor the answer, 4 steps is because it is a lcm model that the lcm lora is mixed it in it. indeed it is strange, maybe i need more tests

**Comment by Amin456789** [comment]
ok it seems the model i was using had problem with chat in this verison, i started to use sdxl lighting 2step q.4.0 with 768x512 and 3-4 steps and it works great

**Comment by Amin456789** [comment]
here's a though i wanted to share for everyone, 
download sdxl turbo gguf from huggingface [there is a q8 that someone uploaded]
in settings of the gui select compress the model so it can make it a quant for usage, use vae fixed too for taesd which is faster
in the koboldcpp = cfg 1, step 1, and choose square or any other, euler a works great 
there u go, u got a low step high quality image generation and very ram friendly, it generate a square image like 35 second in my laptop with cpu and 6gb ram only
i use h2o danube 1.8b model q4  for txt generation, it is fast and much much better than lots of 7bs  i tested, reasoning and everything is very very good for its size, its magical
